### DISCUSSION TITLE
Changing the theia web path

### DISCUSSION THREAD

**Discussion by UltraBlackLinux** [eclipse-theia/theia#13464] [Q&A] [answered]
Hey there,
I'd like to host theia on some.domain/theia/... but the documentation is scarse and doesn't even properly cover how to host this the correct way. Any idea how you can change the path then? I don't wanna deal with subdomains.
Thanks!

**Comment by msujew** [comment, marked as answer]
Hey @UltraBlackLinux,

you would usually solve this using some sort of reverse proxy. Both the subdomain and path version work similarly in the sense that they forward to some `localhost:<port>` address on the container that is used to run Theia. See also [theia-cloud](https://theia-cloud.io/).

Note that I would recommend using subdomains (assuming you allow users to install vscode extensions via the registry view) as otherwise all webviews become security hazards - each vscode extension could potentially read the localStorage/cookie storage of other extensions.

**Reply by UltraBlackLinux** [reply]
well I already figured it out and forgot to close this. Theia is the program that has caused me the least problems in this regard EVER. It has worked straight out of the box without saying a thing. Great tool.

**Reply by yuhorun** [reply]
> well I already figured it out and forgot to close this. Theia is the program that has caused me the least problems in this regard EVER. It has worked straight out of the box without saying a thing. Great tool.

Hello, how did you solve it? Do you have any relevant documents or cases?@UltraBlackLinux

**Reply by UltraBlackLinux** [reply]
I literally forgot lol. All I know now is that I switched to the vscode server some time later because the plugin handling on theia got really really annoying.

**Reply by yuhorun** [reply]
> I literally forgot lol. All I know now is that I switched to the vscode server some time later because the plugin handling on theia got really really annoying.

haha😄
